internal/venice-client-common/src/main/java/com/linkedin/venice/authentication/ClientAuthenticationProviderFactory.java

```diff
@@ -0,0 +1,27 @@
+package com.linkedin.venice.authentication;
+
+import static com.linkedin.venice.CommonConfigKeys.AUTHENTICATION_TOKEN;
+
+import com.linkedin.venice.authentication.jwt.ClientAuthenticationProviderToken;
+import com.linkedin.venice.utils.VeniceProperties;
+import java.util.Map;
+
+
+public final class ClientAuthenticationProviderFactory {
+  private ClientAuthenticationProviderFactory() {
+
+  }
+
+  public static ClientAuthenticationProvider build(VeniceProperties properties) {
+    return build(properties.toProperties());
+  }
+
+  public static ClientAuthenticationProvider build(Map properties) {
```

Comment:
this method exists because in some points we pass something that implements "Map", like org.apache.samza...Config   (implenents Map<String, String)
like here
https://github.com/linkedin/venice/blob/f654de347c97e7808ff8b96037d4a75abdd9eb95/clients/venice-samza/src/main/java/com/linkedin/venice/samza/VeniceSystemFactory.java#L378
and sometimes we pass a Properties object (that is Map<Object, Object>)

We can keep Properties and adapt the code that uses Samza Config, or use here Map<Object, Object>.

I would use Map<String, Object>, but we will have to add some casts/conversions

Reply:
let's keep Properties